.circleci/config.yml

```diff
@@ -0,0 +1,75 @@
+---
+version: 2.0
+
+references:
+  stack_build: &stack_build
+    docker:
+      # https://github.com/haskell-works/stack-build/blob/master/minimal/Dockerfile
+      - image: quay.io/haskell_works/stack-build-minimal
+
+    steps:
+      - checkout
+      - run:
+          name: Digest
+          command: |
+            echo -- "${STACK_ARGUMENTS}" > rdigest
+            git ls-files | xargs md5sum > sdigest
+
+      - restore_cache:
+          keys:
+            - 1-{{ .Branch }}-{{ checksum "rdigest" }}-{{ checksum "sdigest" }}
+            - 1-{{ .Branch }}-{{ checksum "rdigest" }}-
+            - 1-{{ .Branch }}-
+            - 1-master-
+
+      - run:
+          name: Dependencies
+          command: |
+            stack setup $STACK_ARGUMENTS
+            stack build stratosphere $STACK_ARGUMENTS --dependencies-only \
+            --test --no-run-tests --jobs=1 --flag stratosphere:-library-only
+          no_output_timeout: 30m
```

Comment:
Is the 30 minute timeout necessary? Does no output get printed for 30 minutes sometimes? That seems crazy.

Reply:
Yeah, it timed out a few times at 10 minutes. Probably only happens on full rebuilds though.

Reply:
Ohh, I bet that's back when you had `gen` building too. With just `stratosphere` by itself you will have a line of stdout per module. I'm 99% sure you can remove these now.

Reply:
Ah good call, that's true. Doc generation takes a while but that's alright.